### PR TITLE
added typecheck for keytable

### LINKED ISSUE #1829: typecheck keytable

### PR DESCRIPTION
**Pull request by jigold**
Resolves #1829

### CODE REVIEW

File: python/hail/keytable.py [hunk 1]
```diff
@@ -74,6 +74,11 @@ def __repr__(self):
         return self._jkt.toString()
 
     @staticmethod
+    @handle_py4j
+    @typecheck(hc=HailContext,
+               rows_py=listof(anytype),
```
Comment (tpoterba) [inline python/hail/keytable.py]: rows_py should be a list of Struct

schema should be a TStruct

File: python/hail/keytable.py [hunk 2]
```diff
@@ -844,6 +891,7 @@ def persist(self, storage_level="MEMORY_AND_DISK"):
 
         return KeyTable(self.hc, self._jkt.persist(storage_level))
 
+    @handle_py4j
     def order_by(self, *cols):
```
Comment (tpoterba) [inline python/hail/keytable.py]: should be able to typecheck this guy

File: python/hail/keytable.py (hunk 1 above)
Comment (tpoterba) [inline python/hail/keytable.py]: Leaving out a param from typecheck raises a `RuntimeError`, which is what we want (and why your test failed I think)

**Comment by tpoterba** [comment]
need to import struct

**Comment by jigold** [comment]
@tpoterba Was this the behavior you were expecting for Struct? ```TypeError: from_py: parameter 'rows_py': expected list[hail.representation.annotations.Struct], found list: '[{'a': 5}, {'a': 5, 'b': 'quam'}, {'a': -1, 'b': 'quam'}, {'b': 'foo'}, {'a': 7, 'b': 'baz'}]'```

**Comment by tpoterba** [comment]
the error message? or what?

Can you write a docstring for that function?

**Comment by tpoterba** [comment]
It looks like TStruct._convert_to_j accepts dicts or structs due to duck typing. Is this intentional @cseed?

**Comment by jigold** [comment]
@tpoterba back to you.

File: python/hail/keytable.py [hunk 3]
```diff
@@ -74,6 +75,12 @@ def __repr__(self):
         return self._jkt.toString()
 
     @staticmethod
+    @handle_py4j
+    @typecheck(hc=anytype,
+               rows_py=oneof(listof(Struct), listof(dictof(strlike, anytype))),
```
Comment (tpoterba) [inline python/hail/keytable.py]: I think this is right.

**Pull request merged**